libafl/src/executors/inprocess.rs

```diff
@@ -1285,12 +1290,34 @@ impl InChildProcessHandlers {
     {
         unsafe {
             let data = &mut FORK_EXECUTOR_GLOBAL_DATA;
-            child_signal_handlers::setup_child_panic_hook::<E, I, OT, S>();
+            // child_signal_handlers::setup_child_panic_hook::<E, I, OT, S>();
+            setup_signal_handler(data)?;
+            compiler_fence(Ordering::SeqCst);
+            Ok(Self {
+                crash_handler: child_signal_handlers::child_crash_handler::<E, I, OT, S>
+                    as *const c_void,
+                timeout_handler: ptr::null(),
+            })
+        }
+    }
+
+    /// Create new [`InChildProcessHandlers`].
+    pub fn with_timeout<E, I, OT, S>() -> Result<Self, Error>
```

Comment:
Maybe the timeout executor wasn't a good idea to begin with? We need a special handling for almost every executor..